packages/ui/src/modal-side/ModalSide.tsx

```diff
@@ -10,31 +10,191 @@ import {
 } from "react-aria-components";
 
 import { TESTID_MODAL_SIDE } from "@repo/constants/src/testids";
-
 import "./ModalSide.css";
 import Icon from "../icon/Icon";
 import Button from "../button/Button";
 import { parseStringToComponents } from "web/utils/string";
+import {
+  ModalSideDataEnum,
+  ArticleType,
+  SubClauseType,
+  GlossaryContentType,
+  ArticleContentType,
+} from "@repo/data/useGlossaryData";
+import { DEFINED_TERMS_SECTION_NUMBER } from "@repo/constants/src/constants";
 
 export interface ModalSideProps {
+  type:
+    | typeof ModalSideDataEnum.GLOSSARY
+    | typeof ModalSideDataEnum.BUILDING_CODE;
   triggerContent: ReactNode;
-  sections: { number: string; header: string; content: string }[];
-  scrollToSection?: string;
+  modalData: ArticleType[];
+  scrollTo?: string;
   "data-testid"?: string;
 }
 
+function renderDefinitionList(
+  definitionList: string[],
+  customHandler?: (location: string) => void,
+) {
+  return (
+    <ul className="ui-ModalSide--BuildingCodeList">
+      {definitionList.map((item, index) => (
+        <li key={index}>{parseStringToComponents(item, customHandler)}</li>
+      ))}
+    </ul>
+  );
+}
+
+function renderSubClauses(subClauses: string[]) {
+  return (
+    <ol type="i" className="ui-ModalSide--BuildingCodeList">
+      {subClauses.map((subClause, index) => (
+        <li key={index}>{subClause}</li>
+      ))}
+    </ol>
+  );
+}
+
+function renderSubsections(subsections: SubClauseType[]) {
+  return (
+    <ol type="a" className="ui-ModalSide--BuildingCodeList">
+      {subsections.map((subsection, index) => (
+        <li key={index}>
+          {subsection.description}
+          {subsection.subClauses && renderSubClauses(subsection.subClauses)}
+        </li>
+      ))}
+    </ol>
+  );
+}
+
+// TODO: (ANY) Move to GlossaryContent component
```

Comment:
When do we anticipate this being addressed?

Reply:
It's not mandatory but I think it would be good to add at some point maybe in the polish tickets once the modal is done being changed. I had to stop writing code at some point since my ticket was only about Glossary DT hahaa